### ISSUE TITLE
Less simplistic splitting for edit:last-cmd

### ISSUE THREAD

**Issue by hanche** [elves/elvish#1491]
If I want to reuse an argument from the middle of the previous command, I hit Meta-comma. But if that argument contains spaces, I can't get it as a single unit (see example below). This in contrast with Meta-period, which always gets the last argument, spaces and all. This could be improved upon, quite a lot I think.

```
⬥ grep foo 'file with spaces  in the filename' > blah
⬥ grep -c foo← here I hit Meta-comma, with the result below
 LASTCMD
    grep foo 'file with spaces  in the filename' > blah
  0 grep
  1 foo
  2 'file
  3 with
  4 spaces
  5 in
  6 the
  7 filename'
  8 >
  9 blah
```

What's more, with complex commands it would be nice to be able to drill down into the distinct arguments. To explain what I mean, assume the previous command was

```
⬥ foo (bar {|x y| put [$y $x]) 'abc def' # silly example
```

and I hit Meta-command, I would like to see the completion list

```
  0 foo (bar {|x y| put [$y $x]) 'abc def' # silly example
  1 foo
  2 (bar {|x y| put [$y $x]) 
  3 'abc def'
  4 # silly example
```

and now, if I choose item 2, I would want the list to be replaced by

```
  0 (bar {|x y| put [$y $x]) 
  1 bar
  2 {|x y| put [$y $x])
```

with further drilling down if I were to choose item 2 again. This “drilling down” could also be applied to lists, braced lists, functions, perhaps even concatenations such as `(styled …)(styled …)` or `a{b c d}`.

I have no idea how hard this would be to implement, but one can always wish. 😉

**Comment by aca** [comment]
```bash
$ echo "hello world"
hello world
$ echo $_
hello world
```
fish `history-token-search-backward`
 to match argument from previous command

**Comment by aca** [comment]
Seems like there's already todo comment on this
https://github.com/elves/elvish/blob/2c8bae8496bc0cd5ccc128646926923ad2deee7e/pkg/edit/listing.go#L86

**Comment by krader1961** [comment]
@hanche: I agree the current behavior is suboptimal but the devil is in the details. What is an "argument" in the context of exposing the tokens of the "last command"?  Consider a silly, but seemingly trivial, example such as this:

```
> put (put $pwd"/tmp")
▶ /Users/krader/tmp
# Press [Alt-,]
 LASTCMD
    put (put $pwd"/tmp")
  0 put
  1 (put
  2 $pwd"/tmp")
```

What should the "last command" tokens be in that example? Should it instead be the entire command line plus `["put", "(", "put", "$pwd"/tmp", "$pwd", "/tmp", ")"]`? Something else?

**Comment by hanche** [comment]
> I don't think it should remove parens from either level.

On further thought, I guess I agree. My reason for suggesting it was based on convenience, but it would violate POLA. And the convenience argument isn't *that* strong: In the most likely situation where you want the body of a parenthesised expression, you are likely to insert it into an empty command line, where removing the parens is just a quick `C-a C-d C-e backspace` away.
